client/packages/coldchain/src/Equipment/DetailView/AppBarButtons.tsx

```diff
@@ -4,54 +4,67 @@ import {
   Grid,
   useTranslation,
   PrinterIcon,
-  ReportContext,
-  LoadingButton,
-  useUrlQueryParams,
+  ButtonWithIcon,
+  useNotification,
+  useDisabledNotification,
 } from '@openmsupply-client/common';
+
 import { useAssets } from '../api';
-import {
-  ReportRowFragment,
-  ReportSelector,
-  useReport,
-} from '@openmsupply-client/system';
 import { UpdateStatusButton } from './UpdateStatusButton';
-import { JsonData } from '@openmsupply-client/programs';
+import { Environment } from '@openmsupply-client/config';
 
 export const AppBarButtonsComponent = () => {
   const { data } = useAssets.document.get();
-  const t = useTranslation('common');
-  const { print, isPrinting } = useReport.utils.print();
-  const {
-    queryParams: { sortBy },
-  } = useUrlQueryParams();
+  const t = useTranslation('coldchain');
+  const { error, success } = useNotification();
+  const { data: settings } = useAssets.utils.labelPrinterSettings();
+  const { show, DisabledNotification } = useDisabledNotification({
+    title: t('heading.unable-to-print'),
+    message: t('error.label-printer-not-configured'),
+  });
+  const onClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if (settings === null) {
+      show(e);
+    } else {
+      printQR();
+    }
+  };
 
-  const printReport = (
-    report: ReportRowFragment,
-    args: JsonData | undefined
-  ) => {
-    if (!data) return;
-    print({
-      reportId: report.id,
-      dataId: data?.id,
-      args,
-      sort: { key: sortBy.key, desc: sortBy.isDesc },
-    });
+  const printQR = () => {
+    fetch(Environment.PRINT_LABEL_QR, {
+      method: 'POST',
+      body: JSON.stringify({
+        code: data?.id,
+        message: `${t('label.serial')}: ${data?.serialNumber ?? ''}\n${t(
+          'label.asset-number'
+        )}: ${data?.assetNumber ?? ''}`,
+      }),
+      credentials: 'include',
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then(response => {
+        if (response.status === 200) {
+          success(t('messages.success-printing-qr'))();
+          return;
+        }
+        error(t('error.printing-qr'))();
+      })
+      .catch(() => {
+        error(t('error.printing-qr'))();
+      });
   };
 
   return (
     <AppBarButtonsPortal>
       <Grid container gap={1}>
         <UpdateStatusButton assetId={data?.id} />
-        <ReportSelector context={ReportContext.Asset} onPrint={printReport}>
-          <LoadingButton
-            variant="outlined"
-            startIcon={<PrinterIcon />}
-            isLoading={isPrinting}
-          >
-            {t('button.print')}
-          </LoadingButton>
-        </ReportSelector>
+        <ButtonWithIcon
```

Comment:
```suggestion
        <LoadingButton
```
It's easy to double click this and print twice, would be good to avoid that...